File: scripts/extract_slides.py
import os
import pypdf

# Define paths
BASE_DIR = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
SLIDE_DIR = os.path.join(BASE_DIR, 'SLIDE')
DATA_DIR = os.path.join(BASE_DIR, 'data')

def extract_text_from_pdf(pdf_path):
    print(f"Processing: {pdf_path}")
    text = ""
    try:
        reader = pypdf.PdfReader(pdf_path)
        for page in reader.pages:
            text += page.extract_text() + "\n"
    except Exception as e:
        print(f"Error reading {pdf_path}: {e}")
    return text

def main():
    if not os.path.exists(SLIDE_DIR):
        print(f"Directory not found: {SLIDE_DIR}")
        return

    pdf_files = [f for f in os.listdir(SLIDE_DIR) if f.lower().endswith('.pdf')]
    print(f"Found {len(pdf_files)} PDF files.")

    for filename in pdf_files:
        pdf_path = os.path.join(SLIDE_DIR, filename)
        extracted_text = extract_text_from_pdf(pdf_path)
        
        if extracted_text:
            output_filename = f"extracted_{filename}.txt"
            output_path = os.path.join(DATA_DIR, output_filename)
            
            with open(output_path, "w", encoding="utf-8") as f:
                f.write(extracted_text)
            
            print(f"Saved text to: {output_path}")

if __name__ == "__main__":
    main()

File: scripts/extract_slides.js
const fs = require('fs');
const path = require('path');
const PDFParser = require('pdf2json');

const slideDir = path.join(__dirname, '../SLIDE');

async function extractSlides() {
    try {
        const files = fs.readdirSync(slideDir).filter(file => file.toLowerCase().endsWith('.pdf'));

        console.log(`Found ${files.length} PDF files.`);

        for (const file of files) {
            const filePath = path.join(slideDir, file);

            console.log(`\n--- Extracting: ${file} ---`);

            const pdfParser = new PDFParser(this, 1); // 1 = text content only

            // Create a promise to handle the async event-based parsing
            const parsePromise = new Promise((resolve, reject) => {
                pdfParser.on("pdfParser_dataError", errData => {
                    console.error(`Error parsing ${file}:`, errData.parserError);
                    resolve(); // Resolve anyway to continue to next file
                });

                pdfParser.on("pdfParser_dataReady", pdfData => {
                    try {
                        const text = pdfParser.getRawTextContent().replace(/\n\s*\n/g, '\n').trim();

                        console.log(`Pages: ${pdfData.formImage.Pages.length}`);
                        console.log(`Content Preview:\n${text.substring(0, 500)}...`);

                        const outputFilename = `extracted_${file}.txt`;
                        const outputPath = path.join(__dirname, '../data', outputFilename);
                        fs.writeFileSync(outputPath, text);
                        console.log(`Full content saved to ${outputPath}`);
                        resolve();
                    } catch (e) {
                        console.error("Error processing text:", e);
                        resolve();
                    }
                });
            });

            pdfParser.loadPDF(filePath);
            await parsePromise;
        }
    } catch (err) {
        console.error("Error reading directory:", err);
    }
}

extractSlides();
